### DISCUSSION TITLE
Remove `md5`, `sha1`, and `sha2` attributes

### DISCUSSION THREAD

**Discussion by rroupski** [ocsf/ocsf-schema#122] [Ideas]
Instead, use `fingerprint` object, which is more flexible and allows to add new algorithms.

The issue with `md5`, `sha1`, and `sha2` attributes is that the algorithm name is used as an attribute name. Adding a new algorithm requires defining a new attribute.

**Comment by magnologan** [comment]
I agree with the approach, but as these hashes are different in size, how are you going to accommodate new algorithms that are larger than the current ones in the future?

**Reply by rroupski** [reply]
The hash is a string so it will not be an issue is if set a large enough length. For example, 1024.

**Reply by irakledibm** [reply]
I would suggest adding an object "hashes" to support different algorithms:

**Reply by irakledibm** [reply]
{
  "caption": "Hashes",
  "name": "hashes",
  "description": "The hashes object containing a hashing algorithms.",
  "extends": "object",
  "attributes": {
    "md_5": {
      "caption": "MD5",
      "description": "MD5 hashing algorithm.",
      "requirement": "optional"
    },
    "sha_1": {
      "caption": "SHA_1",
      "description": "SHA_1 hashing algorithm.",
      "requirement": "optional"
    },
    "sha_256": {
      "caption": "SHA_256",
      "description": "SHA_256 hashing algorithm.",
      "requirement": "optional"
    },
    "sha_512": {
      "caption": "SHA_512",
      "description": "SHA_512 hashing algorithm.",
      "requirement": "optional"
    }
  }
}

**Comment by AWSSecEng** [comment]
Implemented in a935cd4815a171d9301c4f00af70fae2810afa86
